dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackedEntityProgramOwnerService.java

```diff
@@ -43,10 +46,19 @@
 /**
  * @author Ameen Mohamed
  */
-@RequiredArgsConstructor
 @Service("org.hisp.dhis.tracker.acl.TrackedEntityProgramOwnerService")
 public class DefaultTrackedEntityProgramOwnerService implements TrackedEntityProgramOwnerService {
   private final TrackedEntityProgramOwnerStore trackedEntityProgramOwnerStore;
+  private final Cache<OrganisationUnit> ownerCache;
+
+  public DefaultTrackedEntityProgramOwnerService(
+      TrackedEntityProgramOwnerStore trackedEntityProgramOwnerStore, CacheProvider cacheProvider) {
+    checkNotNull(trackedEntityProgramOwnerStore);
+    checkNotNull(cacheProvider);
```

Comment:
```suggestion
```
[`spring injected contstructors do not need nullability annotations`](https://github.com/dhis2/wow-backend/blob/master/guides/code-guide.md#constructors)